artcommon/artcommonlib/runtime.py

```diff
@@ -1,13 +1,75 @@
+import logging
+import os
+import traceback
+import warnings
 from abc import ABC, abstractmethod
 
+from artcommonlib import logutil
+from artcommonlib.konflux.konflux_db import KonfluxDb
+
+
 # an abstract class intended to be used by anything looking for config specific to a single
 # ocp-build-data group.
 
 
 class GroupRuntime(ABC):
+    def __init__(self, **kwargs):
+        self.debug = False
+        self.quiet = False
+        self._logger = None
+        self.debug_log_path = None
+
+        self.working_dir = None
+
+        self.initialized = False
+
+        self.konflux_db = None
+
+        for key, val in kwargs.items():
+            self.__dict__[key] = val
+
+    def initialize(self):
+        self.initialize_logging()
+        self.initialize_konxflux_db()
```

Comment:
Actually I didn't see a strong reason to make it a field in Runtime and initialize KonfluxDB for nearly all art-tools commands. I feel much better to initialize it when we need it, at least for now.

Reply:
Doozer runtime has a `db` field [here](https://github.com/openshift-eng/art-tools/blob/main/doozer/doozerlib/runtime.py#L97). I followed the same pattern, by adding the new `konflux_db` field in the artcommon runtime. Initialization is still optional, as it happens only if we provide the required env vars (exactly the same way it currently happens for legacy DB in doozer). 

I don't have a strong opinions on this, but I generally prefer to have consistent patterns in the code.